ccm_web/client/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ChangeEvent, useRef } from 'react'
+import { Button, Card, CardContent, makeStyles, Typography } from '@material-ui/core'
+import CloudUploadIcon from '@material-ui/icons/CloudUpload'
+import { useSnackbar } from 'notistack'
+
+interface FileUploadProps {
+  onUploadComplete: (file: File) => void
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    textAlign: 'center'
+  },
+  cardContent: {
+    cursor: 'pointer'
+  },
+  actionsBar: {
+    backgroundColor: '#EEEEEE'
+  },
+  actionsBarIcon: {
+    verticalAlign: 'middle'
+  },
+  input: {
+
+  },
+  uploadIcon: {
+    color: '#3F648E'
+  }
+}))
+
+function FileUpload (props: FileUploadProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
+  const handleFileSelected = (file: File|undefined): void => {
+    if (file === undefined) return
+    if (file.name.toUpperCase().match(/.+.CSV/) == null) {
```

Comment:
@chrisrrowland this seem to be looking for `.csv` file. Is that the current restriction we are putting?

Reply:
Instead of looking for a match for `.csv` can we look for the file `type: text/csv` and do validation that way
![Screen Shot 2021-05-06 at 12 11 33 PM](https://user-images.githubusercontent.com/8579775/117331202-84a06980-ae64-11eb-8c42-1fb0b2c4dad0.png)

Reply:
@melindakraft I think we are limiting this file type to `csv` how would you feel about this?

Reply:
I hadn't considered other file extensions since every reference in the text explicitly says CSV, however it's certainly possible to have csv data in a file with a different extension.

Are the files output by canvas not .csv files?  If not that seems like it would be a terrible  restriction :-p

Reply:
Canvas seems to let the download as [CSV](https://canvas-test.it.umich.edu/courses/403334/gradebook). So if you do Actions --> Export then it is downloading in CSV.
After downloading the file from here the user goes to Faculty tools and they seems to support 2 format csv, excel. https://csprod.dsc.umich.edu/htmldoc/eng/dftie/lsaa/htm/CU_SGRD_Upload.html

I think to me it makes sense to keep it like that. I talked to Dan as well and he was fine with supporting CSV files only.

Reply:
Ok, You can leave the check as regex  `CSV`, if you believe file type might be excel and still the file extension as .csv

Reply:
Well the file extension is really irrelevant, its type of data in the file that matters, right?
It's not going to work with an excel file that just happens to have a .csv extension, though that would require someone manually changing an extension.

Reply:
I'd wager that there are javascript excel libraries out there, so if we wanted to handle excel files we could tackle that.

Reply:
Currently Canvas seems to download the gradebook as csv file and I would say it is fine. Speaking to Dan was also OK to look for CSV file and support that only. 

If we have to support excel it won't be hard, just don't know if that is the real usecase. we might end up supporting multiple OS type Sheets support

Reply:
Resolving